webapp/channels/src/components/emoji/emoji_page.tsx

```diff
@@ -25,62 +25,52 @@ type Props = {
     };
 }
 
-export default class EmojiPage extends React.PureComponent<Props> {
-    static defaultProps = {
-        teamName: '',
-        teamDisplayName: '',
-        siteName: '',
-    };
-
-    componentDidMount() {
-        this.updateTitle();
-        this.props.actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
+export default function EmojiPage({teamDisplayName = '', teamName = '', siteName = '', scrollToTop, currentTheme, actions}: Props) {
+    useEffect(() => {
+        updateTitle();
+        actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
         Utils.resetTheme();
-    }
 
-    componentWillUnmount() {
-        Utils.applyTheme(this.props.currentTheme);
-    }
+        return () => {
+            Utils.applyTheme(currentTheme);
+        };
+    }, [actions, currentTheme]);
 
-    updateTitle = () => {
-        document.title = Utils.localizeMessage('custom_emoji.header', 'Custom Emoji') + ' - ' + this.props.teamDisplayName + ' ' + this.props.siteName;
-    };
+    useEffect(() => {
+        updateTitle();
+    }, [siteName]);
 
-    componentDidUpdate(prevProps: Props) {
-        if (this.props.siteName !== prevProps.siteName) {
-            this.updateTitle();
-        }
-    }
+    const updateTitle = () => {
+        document.title = Utils.localizeMessage('custom_emoji.header', 'Custom Emoji') + ' - ' + teamDisplayName + ' ' + siteName;
```

Comment:
We are trying to get rid of `Utils.localizeMessage`. Can you use here intl? (`useIntl` and `intl.formatMessage`)